### PR TITLE
Adds input component and its common props

### PR DESCRIPTION
**Pull request by Binabh**

### CODE REVIEW

**Comment by itSubeDibesh** [comment]
@Binabh sir, looks like we are missing some important details for the components as per our 
[Figma](https://www.figma.com/file/eqsVKuX1Q9bQ5SKL5ZhIVi/%40shadcn%2Fui---Design-System-w?type=design&node-id=710%3A22619&mode=design&t=bV2uXoKmzhQWUSvX-1) design. Could you have a look 🙏

**Comment by Binabh** [comment]
@itSubeDibesh Thank you for your review! I have addressed all of the known scenarios and all variants as per figma are now doable. Review Please and suggestions on improvements if any.

File: src/components/atoms/Input/index.tsx [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+      small: "h-9",
+      default: "h-11",
+      large: "h-14",
+    },
+  },
+  defaultVariants: {
+    variant: "default",
+  },
+})
+type InputVarinats = VariantProps<typeof input>;
```
Comment (itSubeDibesh) [inline src/components/atoms/Input/index.tsx]: @Binabh everything else looked good while I tested but got a small typo in `InputVarinats`. Could you please fix this 🙇

Comment (Binabh) [reply src/components/atoms/Input/index.tsx]: Thank you for noticing @itSubeDibesh I have fixed it.

**Review by itSubeDibesh — APPROVED** [review]
LGTM 🚀, Thanks for your quick response 🙇

**Comment by SudeepTimalsina** [comment]
@Binabh Can you please add character count on the text area and input component 🙏 
image

**Comment by Binabh** [comment]
Thank You @SudeepTimalsina . Will work on this feature.

**Comment by balraaz15** [comment]
@Binabh sir, I think we should include the variant for error as well, along with error message display. Also, let's handle composition as well. 🙇

**Review by dipeshdulal — APPROVED** [review]
Merging this for now. 
The remaining changes will be added later on with a separate PR to complete according to the designs in figma.

**Review by SudeepTimalsina — APPROVED** [review]
LGTM  🙇

**Pull request merged**